packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js

```diff
@@ -403,6 +409,23 @@ export const ComplexItem = withTranslation(I18N_DOMAIN_COMPONENTS)(UntranslatedC
 export function Item(props) {
 	const { data, tagged, jsonpath, tupleLabel } = props;
 
+	const [lineItemWidth, setLineItemWidth] = useState(false);
+	const [showIcon, setShowIcon] = useState(false);
+	const [nativeValueWrap, setNativeValueWrap] = useState(false);
+
+	const lineItemRef = useRef();
+
+	useLayoutEffect(() => {
+		const { current } = lineItemRef;
+		if (current) {
+			const ref = current.ref;
+			if (ref.offsetParent.offsetWidth < ref.scrollWidth) {
+				setLineItemWidth(true);
+				setShowIcon(true);
+			}
```

Comment:
this trigger two renders for one action. Don t you think we could display icon if linewith is true ?